File: two_sums.py
# https://leetcode.com/problems/two-sum/

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for item_index, item in enumerate(nums):
            nums[item_index] = 'TEMPORARY_REPLACEMENT'
            ideal_value = target-item
            if ideal_value in nums:
                # found that number
                # return its index
                another_item_index = nums.index(ideal_value)
                return [item_index, another_item_index]
            nums[item_index] = item
                
        return []

File: leetcode/two_sums.go
// https://leetcode.com/problems/two-sum/

package main

import "fmt"

func main() {
	// default values
	numbers := []int{2, 3, 5, 7, 11}
	var target = 8

	// find result
	fmt.Println(twoSum(numbers, target))
}

func twoSum(nums []int, target int) []int {
	numsMap := make(map[int]int)

	for itemIndex, item := range nums {
		numsMap[item] = itemIndex
	}

	for itemIndex, item := range nums {
		idealValue := target - item
		value, ok := numsMap[idealValue]
		if ok && itemIndex != value {
			// value present in the map
			return []int{itemIndex, value}
		}
	}
	return nil
}
